### PR TITLE
[PVR] EPG Search Fixes

### PR DESCRIPTION
**Pull request by ksooo**
Some EPG Search related fixes.

Runtime-tested on macOS and Android, latest Kodi master.

@phunkyfish if/when you find some time for a review...

### CODE REVIEW

File: xbmc/pvr/epg/EpgSearchFilter.cpp [hunk 1]
```diff
@@ -346,6 +361,26 @@ bool CPVREpgSearchFilter::MatchChannel(const std::shared_ptr<CPVREpgInfoTag>& ta
          CServiceBroker::GetPVRManager().Clients()->IsCreatedClient(tag->ClientID());
 }
 
+bool CPVREpgSearchFilter::MatchChannelGroup(const std::shared_ptr<CPVREpgInfoTag>& tag) const
+{
+  if (m_iChannelGroupID != -1)
+  {
+    if (!m_groupIdMatches.has_value())
+    {
+      const std::shared_ptr<CPVRChannelGroup> group = CServiceBroker::GetPVRManager()
```
Comment (phunkyfish) [inline xbmc/pvr/epg/EpgSearchFilter.cpp:370]: I know clang format gives a lot of benefits but this is super ugly!

File: xbmc/pvr/epg/EpgSearchFilter.cpp [hunk 2]
```diff
@@ -346,6 +361,26 @@ bool CPVREpgSearchFilter::MatchChannel(const std::shared_ptr<CPVREpgInfoTag>& ta
...
+                                                          .ChannelGroups()
+                                                          ->Get(m_bIsRadio)
+                                                          ->GetById(m_iChannelGroupID);
```
Comment (phunkyfish) [inline xbmc/pvr/epg/EpgSearchFilter.cpp:373]: I know clang format gives a lot of benefits but this is super ugly!

File: xbmc/pvr/epg/EpgSearchFilter.h [hunk 3]
```diff
@@ -151,11 +156,14 @@ namespace PVR
     // PVR specific filters
     bool m_bIsRadio; /*!< True to filter radio channels only, false to tv only */
     int m_iClientID = -1; /*!< The client id */
+    int m_iChannelGroupID{-1}; /*! The channel group id */
```
Comment (phunkyfish) [inline xbmc/pvr/epg/EpgSearchFilter.h:159]: Why does this have a different initialiser to the other lines?

**Review by phunkyfish — COMMENTED** [review]
Just a query

File: xbmc/pvr/epg/EpgSearchFilter.cpp (hunk 2 above)
Comment (ksooo) [reply xbmc/pvr/epg/EpgSearchFilter.cpp:373]: Maybe I can trick clang-format by using `auto` ?

File: xbmc/pvr/epg/EpgSearchFilter.h (hunk 3 above)
Comment (ksooo) [reply xbmc/pvr/epg/EpgSearchFilter.h:159]: coding guidelines: https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#54-initialization

Comment (phunkyfish) [reply xbmc/pvr/epg/EpgSearchFilter.h:159]: Ok, well you learn something new everyday.

**Review by phunkyfish — APPROVED** [review]

File: xbmc/pvr/epg/EpgSearchFilter.cpp (hunk 2 above)
Comment (ksooo) [reply xbmc/pvr/epg/EpgSearchFilter.cpp:373]: No luck. So we will have to live this, even though I agree it's ugly.

**Pull request merged**